File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
/* 알고리즘
1. participant 배열 정렬
2. completion 배열 정렬
3. 두 배열을  equals 비교 -> 없는 것만 participant 배열에 담아서 출력
*/

import java.util.*;

class Solution {
    
    //static int i;
    
    public String solution(String[] participant, String[] completion) {
        
        // 1. participant 배열 정렬
        Arrays.sort(participant);
        
        // 2. completion 배열 정렬
        Arrays.sort(completion);
        
        int i;
        
        // 3. 두 배열 비교 -> 없는 것만 participant 배열에 담아서 출력
        for(i=0; i<completion.length; i++){
            if(!participant[i].equals(completion[i])){
                return participant[i];
            }
        }
        return participant[i];
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
# 알고리즘
# participant 배열과 completion 배열을 비교해서 -> 일치하지 않는 경우만 출력하도록 구성
# 1. participant 배열 정렬
# 2. completion 배열 정렬
# 3. 일치하지 않는 경우 -> return

def solution(participant, completion):
    participant.sort()
    completion.sort()
    for p, c in zip(participant, completion):
        if p != c:
            return p
    return participant[-1]
        